static/src/javascripts/lib/geolocation.ts

```diff
@@ -0,0 +1,477 @@
+import { getLocale, storage } from '@guardian/libs';
+import type { CountryCode } from '@guardian/libs/dist/cjs/types/countries';
+import config from 'lib/config';
+import { getCookie } from 'lib/cookies';
+import reportError from 'lib/report-error';
+
+const editionToGeolocationMap: Record<string, string> = {
+	UK: 'GB',
+	US: 'US',
+	AU: 'AU',
+};
+
+const editionToGeolocation = (editionKey = 'UK'): string =>
+	editionToGeolocationMap[editionKey];
+
+const countryCookieName = 'GU_geo_country';
+const countryOverrideName = 'gu.geo.override';
+
+let locale: CountryCode | null;
+
+/*
+   This method can be used as a non async way of getting the country code
+   after init has been called. Returning locale should cover all/most
+   of the cases but if a race condition happen or the cookie is not set,
+   we keep fallbacks to cookie or geo from edition.
+*/
+const getCountryCode = (): string => {
+	const pageEdition: string = (config as {
+		get: (arg: string) => string;
+	}).get('page.edition');
+
+	return (
+		locale ??
+		(storage.local.get(countryOverrideName) as string | null) ??
+		(getCookie(countryCookieName) as string | null) ??
+		editionToGeolocation(pageEdition)
+	);
+};
+
+const setGeolocation = (geolocation: CountryCode | null): void => {
+	locale = geolocation;
+};
```

Comment:
Is this ever used?

Reply:
it was there for memoizing but I have inlined it now!